integration/deployment_test.go

```diff
@@ -0,0 +1,155 @@
+//go:build integration
+// +build integration
+
+package integration
+
+import (
+	"testing"
+	"time"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/deployment"
+	"github.com/openshift-kni/eco-goinfra/pkg/namespace"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	namespacePrefix = "ecogoinfra-deployment"
+	containerImage  = "nginx:latest"
+	timeoutDuration = time.Duration(60) * time.Second
+)
+
+func TestDeploymentCreate(t *testing.T) {
+	t.Parallel()
+	client := clients.New("")
+
+	// Create a random namespace for the test
+	randomNamespace := generateRandomNamespace(namespacePrefix)
+
+	// Create the namespace in the cluster using the namespaces package
+	_, err := namespace.NewBuilder(client, randomNamespace).Create()
+	assert.Nil(t, err)
+
+	// Defer the deletion of the namespace
+	defer func() {
+		// Delete the namespace
+		err := namespace.NewBuilder(client, randomNamespace).Delete()
+		assert.Nil(t, err)
+	}()
+
+	var deploymentName = "create-test"
+
+	builder := deployment.NewBuilder(client, deploymentName, randomNamespace, map[string]string{
+		"app": "test",
+	}, corev1.Container{
+		Name:  "test",
+		Image: containerImage,
+		Command: []string{
+			"sleep",
+			"3600",
+		},
+	})
+
+	// Create a deployment in the namespace
+	_, err = builder.CreateAndWaitUntilReady(timeoutDuration)
+	assert.Nil(t, err)
+
+	// Check if the deployment was created
+	builder, err = deployment.Pull(client, deploymentName, randomNamespace)
+	assert.Nil(t, err)
+	assert.NotNil(t, builder.Object)
+}
+
+func TestDeploymentDelete(t *testing.T) {
+	t.Parallel()
+	client := clients.New("")
+
+	// Create a random namespace for the test
+	randomNamespace := generateRandomNamespace(namespacePrefix)
```

Comment:
Let's run all test in the same namespace and hardcode all names

Reply:
Done.  I kept the namespaces separate (for cleanup and separation) but I hardcoded them.